bundles/org.openhab.binding.bluetooth.generic/src/main/java/org/openhab/binding/bluetooth/generic/internal/GenericBluetoothHandler.java

```diff
@@ -84,20 +86,35 @@ public void initialize() {
         readCharacteristicJob = scheduler.scheduleWithFixedDelay(() -> {
             if (device.getConnectionState() == ConnectionState.CONNECTED) {
                 if (resolved) {
-                    for (CharacteristicHandler charHandler : charHandlers.values()) {
-                        if (charHandler.canRead()) {
+                    handlerToChannels.forEach((charHandler, channelUids) -> {
+                        // Only read the value manually if notification is not on.
+                        // Also read it the first time before we activate notifications below.
+                        if (!device.isNotifying(charHandler.characteristic) && charHandler.canRead()) {
                             device.readCharacteristic(charHandler.characteristic);
                             try {
                                 // TODO the ideal solution would be to use locks/conditions and timeouts
-                                // between this code and `onCharacteristicReadComplete` but
+                                // Kbetween this code and `onCharacteristicReadComplete` but
                                 // that would overcomplicate the code a bit and I plan
                                 // on implementing a better more generalized solution later
                                 Thread.sleep(50);
                             } catch (InterruptedException e) {
                                 return;
                             }
                         }
-                    }
+                        if (charHandler.characteristic.canNotify()) {
+                            ChannelUID channelUID = charHandler.getChannelUID(null);
+                            // Enabled/Disable notifications dependent on if the channel is linked.
+                            if (isLinked(channelUID)) {
```

Comment:
```suggestion
                            // Enabled/Disable notifications dependent on if the channel is linked.
                            if (channelUids.stream().anyMatch(this::isLinked)) {
```

Reply:
I tested this change but for some reason does it always evaluate to true in my tests.

Reply:
Well then something must be going on with `isLinked` then. But checking against the `channelUids` is the correct change since `charHandler.getChannelUID(null)` would return the wrong channel sometimes.

Reply:
You are right, `charHandler.getChannelUID(null)` does not work reliably. I added your suggested code with a TODO.